include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdlops_v2r3.hpp

```diff
@@ -874,20 +884,29 @@ struct GridwiseGemm_k0mk1_k0nk1_mn_xdlops_v2r3_ext
             }
         }();
 
+        const auto K0Pad = math::integer_divide_ceil(K0, K0PerBlock) * K0PerBlock;
+        const auto KPad  = K0Pad * K1Value;
+
+        const auto b_grid_desc_kpad_n = transform_tensor_descriptor(
+            b_grid_desc_k_n,
+            make_tuple(make_right_pad_transform(K, KPad - K), make_pass_through_transform(N)),
+            make_tuple(Sequence<0>{}, Sequence<1>{}),
+            make_tuple(Sequence<0>{}, Sequence<1>{}));
+
         if constexpr(GemmSpec == tensor_operation::device::GemmSpecialization::MNPadding)
         {
             return transform_tensor_descriptor(
-                b_grid_desc_k_n,
-                make_tuple(make_unmerge_transform(make_tuple(K0, K1Value)),
+                b_grid_desc_kpad_n,
+                make_tuple(make_unmerge_transform(make_tuple(K0Pad, K1Value)),
```

Comment:
If You prefer, You can use ck::tensor_operation::device::PadTensorDescriptor

Reply:
Nice `tensor_operation`, but it needs refactoring a lot of places. Let us leave it for the next PR.

Reply:
Sure :)